src/add-liquidity/v2/util.ts

```diff
@@ -99,21 +98,21 @@ export function calculateSubsequentAddLiquidity({
   };
 }
 
-export function calculateInitialAddLiquidity(
+/**
+ * @returns the amount of pool tokens that should be issued for the initial add liquidity operation
+ */
+export function calculateV2InitialLiquidityAmount(
   asset1: AssetWithAmountAndDecimals,
   asset2: AssetWithAmountAndDecimals
-) {
+): bigint {
   if (!asset1.amount || !asset2.amount) {
     throw new Error("Both assets are required for the initial add liquidity");
   }
 
   return BigInt(
-    Math.abs(
-      Math.floor(
-        Math.sqrt(
-          convertToBaseUnits(asset1.decimals, Math.floor(Number(asset1.amount))) *
-            convertToBaseUnits(asset2.decimals, Math.floor(Number(asset2.amount)))
-        ) - V2_LOCKED_POOL_TOKENS
+    Math.floor(
+      Math.abs(
+        Math.sqrt(Number(asset1.amount) * Number(asset2.amount)) - V2_LOCKED_POOL_TOKENS
```

Comment:
This seems to solving the issue. 
But one thing that I didn't understand in general is: we usually use bigint as the default type. But in most of the functions, we convert them to `Number(..)`. I don't really get this. cc @edizcelik

Reply:
This is still confusing for me too. It seems unsafe to use Numbers and maybe we should go over all computations at some point and try to convert to using BigInt where possible.